### PR TITLE
Fix/downstream

### PR DESCRIPTION
**Pull request by KarahanS**
This PR includes the following corrections:
* There was an error with the downstream monitoring - fixed now.
* An error due to the incorrect indentation in self-supervised pretraining - fixed.
* Model names and checkpoints are saved with the timestamps now. 
* Now we can run downstream tasks without logging as well.
* SimCLR hyperparameters are updated to include encoder information.
* Cosine scheduler period is set to `max_epoch` (without restarts).

### CODE REVIEW

**Review by OguzAtaCal — APPROVED** [review]
Looks good to me
